bundles/pcmumlclass/tools.vitruv.applications.pcmumlclass/src/tools/vitruv/applications/pcmumlclass/pcm/PcmRepository.reactions

```diff
@@ -83,11 +84,27 @@ routine createUmlRepositoryPackage(pcm::Repository pcmRepo) {
 			}
 			
 			umlRootModel.nestedPackages += umlRepositoryPkg
+			ensureUmlModelCorrespondenceExists(umlRootModel)
 		}
 		add correspondence between pcmRepo and umlRepositoryPkg tagged with TagLiterals.REPOSITORY_TO_REPOSITORY_PACKAGE
 	}
 }
 
+routine ensureUmlModelCorrespondenceExists(uml::Model newModel) {
+	match {
+		val alreadyCorrespondingModels = retrieve many uml::Model corresponding to UMLPackage.Literals.MODEL
+	}
+	action {
+		execute if (!alreadyCorrespondingModels.contains(newModel)) addUmlModelCorrespondence(newModel)
+	}
+}
+
+routine addUmlModelCorrespondence(uml::Model newModel){
+	action {
+		add correspondence between UMLPackage.Literals.MODEL and newModel
+	}
+}
+
```

Comment:
This is the only change that was necessary to fix the problem in my specific instance, as in my case PCM to UML was executed first, and therefore the UML root model was created here without the correspondence.